modules/eks/node-group.tf

```diff
@@ -51,6 +51,19 @@ resource "aws_security_group_rule" "node" {
   )
 }
 
+resource "aws_security_group_rule" "private_link" {
```

Comment:
Instead of a new sgr can it be baked onto https://github.com/dominodatalab/terraform-aws-eks/blob/main/modules/eks/main.tf#L43

Reply:
I remember why I didnt use that, that rule is conditional for private link

Reply:
I believe you can leverage the same conditional when adding it to `local.node_security_group_rules`,like this 
```hcl
node_security_group_rules = {
    private_link = var.privatelink.enabled ? {
        security_group_id        = aws_security_group.eks_nodes.id
        protocol                 = "tcp"
        from_port                = 30000
        to_port                  = 32767
        type                     = "ingress"
        description              = "NLB from private link to node ports"
        source_security_group_id = try(module.privatelink[0].info.nlb_sg, null)
    } : {}
  }
```

Reply:
I could use merge to add the condition object if you agree

Reply:
yes, that should be fine. btw the idea is just to have these in one place